mitxgraders/stringgrader.py

```diff
@@ -43,34 +128,75 @@ def check_response(self, answer, student_input, **kwargs):
             answer (str): The answer to compare to
             student_input (str): The student's input passed by edX
         """
-        expect = answer['expect']
-        student = student_input
```

Comment:
I'm going to push back on this one. The purpose of writing functions is typically so that you can reuse those functions elsewhere. In this case, all those functions would be called exactly once. Furthermore, most of those functions need to be able to return a grade, which is complicated by being in a subfunction. Finally, if you dropped the comments and didn't count all the lines associated with setting up feedback messages, this isn't really a long function at all. I'll do some refactoring to make it look shorter, and then see if you still feel that way.

Reply:
> The purpose of writing functions is typically so that you can reuse those functions elsewhere. 

Well, I agree that’s a great use of functions. But I would also suggest that a function with a readable, meaningful name that replaces a block of code is a good use, too, even if that function is only called once. For me, it makes the composite function’s structure much faster to understand. 

But, to each their own :) I’m OK reading long functions as long as you’re OK reading my single use functions 😛

Reply:
I think I've substantially improved the readability and flow of the method. Let me know what you think.